### ISSUE TITLE
Cross compilation 2.11.11 and 2.12.3 with SBT 1.0.0

### ISSUE THREAD

**Issue by mr-git** [sbt/sbt#3422]
```
sbt "+all test packagedArtifacts"
```

Doesn't work any more after changing from 0.13.16 to 1.0.0 in `build.properties`. Error log:

```
$ sbt "+all test packagedArtifacts"
[info] Loading settings from idea.sbt ...
[info] Loading global plugins from /Users/mareks/.sbt/1.0/plugins
[info] Loading settings from plugins.sbt ...
[info] Loading project definition from /Users/mareks/work/git/ids/project
[info] Loading settings from build.sbt,version.sbt ...
[info] Set current project to ids (in build file:/Users/mareks/work/git/ids/)
[error] Expected configuration
[error] Expected ':' (if selecting a configuration)
[error] Expected key
[error] Not a valid key: all
[error] ++  2.12.3 project/all test packagedArtifacts
[error]       
```

In addition, we have alias added to project, like:
```
val alias: Seq[sbt.Def.Setting[_]] =
  addCommandAlias("build", "all test packagedArtifacts")
```

Which we used to make cross compilation by lauching `SBT +build` - returns error too:
```
$ sbt +build
[info] Loading settings from idea.sbt ...
[info] Loading global plugins from /Users/mareks/.sbt/1.0/plugins
[info] Loading settings from plugins.sbt ...
[info] Loading project definition from /Users/mareks/work/git/ids/project
[info] Loading settings from build.sbt,version.sbt ...
[info] Set current project to ids (in build file:/Users/mareks/work/git/ids/)
[error] Expected ':' (if selecting a configuration)
[error] Not a valid key: build (similar: loadedBuild)
[error] ++  2.12.3 project/build
[error]   
```

Is this a bug or is there some other way how to execute cross-compilation in SBT 1.0.0?

**Comment by eed3si9n** [comment]
Seems like an error on the new `+` implementation.

**Comment by dwijnand** [comment]
`+` not accepting commands or aliases is known and correct. See https://github.com/sbt/sbt/issues/2776.

Not a bug, imho. But we should see if there are better ways to communicate the error to the user.

**Comment by mr-git** [comment]
so.. there is no real way to make cross-building any more?
Please suggest the "correct" way to make cross-compilation that would be analog to having alias `build` with tasks/commands `all test packagedArtifacts` and calling `+build`.

With 0.13.16 it is possible to execute several tasks in parallel by using `all` - this significantly speeds up compilation, testing and packaging in big multi-module project - as some modules can be packaged and ready as dependecy for next project, while still being unit-tested. Sadly, `all` is not a task, as it looks, at least I didn't find it in `Keys.scala`.

**Comment by mr-git** [comment]
`sbt "; ++2.12.3; build; ++2.11.11; build"` - doesn't look nice...
At least I can hide this behind ugly alias, like:
```
val alias: Seq[sbt.Def.Setting[_]] =
  addCommandAlias("build", "all test packagedArtifacts") ++
  addCommandAlias("cross-build", "; ++ 2.11.11; build; ++ 2.12.3; build")
```
And call `sbt cross-build`...
Is this the new way? At least on http://www.scala-sbt.org/1.0/docs/Cross-Build.html I didn't find any better suggestion :(

**Comment by mr-git** [comment]
Could you please provide some sample how to use `Def.sequential` to run cross-compilation?
I do not understand how to do that from https://github.com/sbt/sbt/issues/2776#issuecomment-311627935

**Comment by eed3si9n** [comment]
It was a design tradeoff (partial cross building using crossScalaVersions vs support for commands), but it's a breaking change.

If you want `all test packagedArtifacts` behavior (run both of those in parallel), the default composition for tasks should work.

```scala
lazy val build = taskKey[Unit]("build")

build := {
  val _ = ((test in Test).value, packagedArtifacts.value)
  ()
}
```

**Comment by mr-git** [comment]
ok, this workaround works

**Comment by eed3si9n** [comment]
Also it looks like we'll support cross building with command soon since @jroper has provided a fix here - https://github.com/sbt/sbt/pull/3446

**Comment by mr-git** [comment]
Nice!
